tests/test_ssl.py

```diff
@@ -4212,3 +4223,57 @@ def client_callback(*args):  # pragma: nocover
 
         with pytest.raises(TypeError):
             handshake_in_memory(client, server)
+
+
+class TestDTLS(object):
+    def test_it_works_at_all(self):
+        s_ctx = Context(DTLS_METHOD)
+
+        def generate_cookie(ssl):
+            return b"xyzzy"
+
+        def verify_cookie(ssl, cookie):
+            return cookie == b"xyzzy"
+
+        s_ctx.set_cookie_generate_callback(generate_cookie)
+        s_ctx.set_cookie_verify_callback(verify_cookie)
+
+        s_ctx.use_privatekey(load_privatekey(FILETYPE_PEM, server_key_pem))
+        s_ctx.use_certificate(load_certificate(FILETYPE_PEM, server_cert_pem))
+
+        s_sock = socket_any_family(type=SOCK_DGRAM)
+        s_sock.bind((loopback_address(s_sock), 0))
+        s = Connection(s_ctx, s_sock)
+        s.set_accept_state()
+
+        c_sock = socket(s_sock.family, type=SOCK_DGRAM)
+        c = loopback_client_factory(c_sock, DTLS_METHOD)
+
+        c_sock.connect(s_sock.getsockname())
+        s_sock.connect(c_sock.getsockname())
+
+        def s_handler():
+            s.DTLSv1_listen()
+            s.do_handshake()
+            s.write(b"hello")
+            assert s.read(100) == b"goodbye"
+            return "ok"
+
+        def c_handler():
+            c.do_handshake()
+            assert c.read(100) == b"hello"
+            c.write(b"goodbye")
+            return "ok"
+
+        with ThreadPoolExecutor(max_workers=2) as executor:
+            s_fut = executor.submit(s_handler)
+            c_fut = executor.submit(c_handler)
+
+            assert s_fut.result() == "ok"
+            assert c_fut.result() == "ok"
+
+        # Check that the MTU set/query functions are doing *something*
+        c.set_ciphertext_mtu(1000)
+        assert 500 < c.get_cleartext_mtu() < 1000
```

Comment:
This test will fail on OpenSSL 1.1.0 due to the notimplemented path in get_cleartext_mtu. Do we need to add a skip?